bazel/linux/templates/init.template.sh

```diff
@@ -1,7 +1,11 @@
 #!/bin/sh
 
 echo ========= {message} - {target} ==========
-trap "poweroff -f" EXIT
+on_exit() {
+    echo $? > "$EXIT_STATUS_FILE" || true
```

Comment:
when `qemu_run` or `qemu_test` is executed, multiple tests can be specified in a single run.

I think we should save the status of each test, ideally also how long each step took (so we can convert those files easily to junit.xml at the end of the run, and show nice break down by test - assuming also that the same 'test script' will run from different 'test targets').

Proposal:
1) Change the {commands} below to have a `run command ...` that saves the status per command?

Question: should we remove the `set -e` global, let all tests run before shutting down the VM? (each  script can have an internal `set -e`? and can call poweroff -f - or a function exported by this script - to stop the test?)

Reply:
I am not sure how we can do that cleanly with what we've got. Some of the commands are initialization commands (e.g. setup mounts). As you pointed out offline, we should split the `commands` into `setup` and `run` phases.

Reply:
Reworked this PR. It's only job is to pass the exit status of the last command from inside the emulator to the host.